endpoint.cpp

```diff
@@ -280,20 +301,50 @@ namespace udpdiscovery {
       void receive() {
         sockaddr_in from_addr;
         AddressLenType addr_length = sizeof(sockaddr_in);
+        bool received_packet = false;
 
-        int length = (int) recvfrom(sock_, &buffer_[0], buffer_.size(), 0, (struct sockaddr *) &from_addr, &addr_length);
-        if (length <= 0)
-          return;
+        int length = (int) recvfrom(binding_sock_, &buffer_[0], buffer_.size(), 0, (struct sockaddr *) &from_addr, &addr_length);
+        if (length <= 0) {
+          length = (int) recvfrom(sock_, &buffer_[0], buffer_.size(), 0, (struct sockaddr *) &from_addr, &addr_length);
```

Comment:
Why trying to receive also from the sending socket? It should receive nothing now on `sock_` AFAIU

Reply:
I though that there can be some sort of direct communication between peers and that is why I tried to receive from sock_. But it is not good to poll this socket, it adds latency, complexity. Removed.